### PR TITLE
Support "flutter refresh" in the Gradle-based sample project

### PR DESCRIPTION
**Pull request by jason-simmons**
After deploying and launching the APK, the app's Dart code can be updated with:
- cd app/src/flutter
- flutter refresh --activity=com.example.flutter/.ExampleActivity

### CODE REVIEW

**Comment by jason-simmons** [comment]
@eseidelGoogle @yjbanov

**Comment by yjbanov** [comment]
LGTM!

File: examples/hello_android/app/src/main/java/com/example/flutter/ExampleActivity.java [hunk 1]
```diff
@@ -76,6 +77,16 @@ protected void onResume() {
         flutterView.onResume();
     }
 
+    @Override
+    protected void onNewIntent(Intent intent) {
+        // Reload the Flutter Dart code when the activity receives an intent
+        // from the "flutter refresh" command.
+        if (Intent.ACTION_RUN.equals(intent.getAction())) {
```
Comment (eseidelGoogle) [inline examples/hello_android/app/src/main/java/com/example/flutter/ExampleActivity.java]: I doubt we want this ability in production code.  Unclear if this should use ACTION_RUN or not:
http://developer.android.com/reference/android/content/Intent.html#ACTION_RUN

Seems like any other process on the phone could hijack the permissions of this app this way. @abarth

Comment (abarth) [inline examples/hello_android/app/src/main/java/com/example/flutter/ExampleActivity.java]: Maybe add a comment about that issue?

**Comment by jason-simmons** [comment]
Made refresh support conditional on the application's debuggable flag.
Refresh will be disabled in release builds of an APK.

**Comment by abarth** [comment]
Nice.  LGTM

**Pull request merged**